### PR TITLE
run: add --timeout option to set maximum service runtime

### PR DESCRIPTION
**Pull request by davide125**
This adds an option to `systemd-run` to set `RuntimeMaxSec`, to make it easier to set an upper bound to the service execution time. Combined with `--wait`, this lets us leverage `systemd-run` to execute processes from Chef in a constrained environment and ensure they're terminated properly.

This PR also includes updates to the completions and adds `image.raw` (from `mkosi`) to `.gitignore`). Happy to split those in separate commits if you prefer.

### CODE REVIEW

**Comment by evverx** [comment]
I think `RuntimeMaxSec` != `timeout`.  For example, `RuntimeMaxSec` doesn't include activation time:

``` sh
-bash-4.3# time systemd-run --wait --timeout=10s --service-type=notify sh -x -c 'sleep 20'
Running as unit: run-u5.service
Finished with result: success
Main processes terminated with: code=exited/status=0
Service runtime: 20.218s

real    0m20.599s
user    0m0.112s
sys     0m0.040s
```

Why not to use `--property RuntimeMaxSec=...`?

**Comment by davide125** [comment]
Ah, good point. I was originally planning to use `TimeoutStartSec` for this, but couldn't get it to work, that's why I wrote it to use `RuntimeMaxSec` instead. Yes, you could set the property directly, but I think this is a useful feature to expose (and it's not very clear from the docs which properties work and which don't in the first place).

File: .gitignore [hunk 1]
```diff
@@ -36,6 +36,7 @@
 /exported
 /exported-*
 /hostnamectl
+/image.raw
```
Comment (martinpitt) [inline .gitignore:39]: This is unrelated, please drop from this patch.

File: shell-completion/zsh/_systemd-run [hunk 2]
```diff
@@ -57,4 +57,6 @@ _arguments \
         '--on-unit-inactive=[Run after SEC seconds from the last deactivation]:SEC' \
         '--on-calendar=[Realtime timer]:SPEC' \
         '--timer-property=[Set timer unit property]:NAME=VALUE' \
+        '--timeout=[Set maximum time for the service to run]:timeout' \
+        '--wait=[Wait until service stopped again]' \
```
Comment (martinpitt) [inline shell-completion/zsh/_systemd-run:61]: Would be nice to have a separate commit for adding the recent `--wait` to the completion. (Same for the bash completion above).

**Comment by martinpitt** [comment]
There are dozens of properties you can set in `[Service]` -- adding new options for all of them sounds like too much unnecessary overhead to me, and also makes the correspondence between those options and the underlying properties less obvious.

**Comment by davide125** [comment]
Fair enough, I'll close this and send a separate PR for the completions stuff.
